### PR TITLE
issues in utilities/refactor of type-checking usage, "bidder" conflict with documentation

### PR DESCRIPTION
**Pull request by nickjacob**
* move yieldbot templating into utilities
* standardize use of `isA` and helper functions
* bug with `hasOwnProperty` not being defined in utilities
* set `bidder` on bidResponse from `bidderCode` to match docs here -- http://prebid.org/publisher-api.html#configure-ad-server-targeting
* typo in example in readme for script tag

### CODE REVIEW

File: src/bidmanager.js [hunk 1]
```diff
@@ -203,34 +184,35 @@ function getKeyValueTargetingPairs(bidderCode, custBidObj) {
...
+
+  utils._each(targeting, function (kvPair) {
+    var key = kvPair.key,
+        value = kvPair.val;
+
+    if (utils.isFn(value)) {
+      try {
+        keyValues[key] = value(custBidObj);
+      } catch (e) {
+        utils.logError("invalid function during k/v targeting", "BIDMANAGER");
```
Comment (mkendall07) [inline src/bidmanager.js]: I like sending the "class" with the error message. Maybe we should add a 3rd param in `utils.logError()` for Exception so we don't lose that info?

File: src/utils.js [hunk 2]
```diff
@@ -264,27 +266,17 @@ exports.getPriceBucketString = function(cpm) {
 };
 
 exports.mapForEach = function(obj, func) {
```
Comment (mkendall07) [inline src/utils.js]: We can probably deprecate this function as I think it is basically doing the same as `utils.each()`

File: src/utils.js [hunk 3]
```diff
@@ -24,18 +26,18 @@ var t_Arr = 'Array',
...
-			var keyString = token + key.toUpperCase() + token;
-			if (typeof map[key] === objectType_undefined) {
-				map[key] = '';
-			}
-			var re = new RegExp(keyString, 'g');
-			str = str.replace(re, map[key]);
-		}
-	}
-	return str;
+  var regex = getRegexpForToken(token);
```
Comment (mkendall07) [inline src/utils.js]: There seems to be some issues with this function. It's not working the way it was previously. Can you revert it for now and make some test cases.

File: src/bidmanager.js (hunk 1 above)
Comment (nickjacob) [inline src/bidmanager.js]: yeah good idea I'll add that

File: src/utils.js (hunk 2 above)
Comment (nickjacob) [inline src/utils.js]: yeah I think it's only used in bidManager, so I'll change that and remove this

File: src/utils.js (hunk 3 above)
Comment (nickjacob) [inline src/utils.js]: yeah I was thinking it'd be good to have some unit tests at least for this file. I'll make those changes, open a different branch with unit tests in it

**Pull request merged**